shoop/core/models/_base.py

```diff
@@ -47,28 +47,37 @@ class Meta:
         abstract = True
 
 
-class ImmutableMixin(object):
+class ChangeProtected(object):
+    protected_fields = None
     unprotected_fields = []
-    immutability_message = _("Cannot change immutable object that is in use")
+    change_protect_message = _("Following fields cannot be changed")
 
     def clean(self, *args, **kwargs):
-        super(ImmutableMixin, self).clean(*args, **kwargs)
+        super(ChangeProtected, self).clean(*args, **kwargs)
         if self.pk:
-            if self._has_any_protected_field_changed() and self._is_in_use():
-                raise ValidationError(self.immutability_message)
+            changed_protected_fields = self._get_changed_protected_fields()
+            if changed_protected_fields and self._is_in_use():
+                message = "{change_protect_message}: {fields}".format(
+                    change_protect_message=self.change_protect_message,
+                    fields=", ".join(sorted(changed_protected_fields)),
+                )
+                raise ValidationError(message)
 
     def save(self, *args, **kwargs):
         self.clean()
-        super(ImmutableMixin, self).save(*args, **kwargs)
+        super(ChangeProtected, self).save(*args, **kwargs)
 
     def _is_in_use(self):
```

Comment:
It's an underscore-prefixed function, which hints that it's a private API, but subclasses of `ChangeProtected`s are supposed to override it. I think it should be just `is_in_use` (and it should have a proper docstring too).